File: backend/app/schemas/system.py
"""System metadata schema definitions."""

from __future__ import annotations

from datetime import datetime
from enum import Enum
from pydantic import BaseModel, ConfigDict


class DatasetServingStatus(str, Enum):
    ACTIVE = "ACTIVE"
    NOT_INGESTED = "NOT_INGESTED"
    INGESTED_UNVERIFIED = "INGESTED_UNVERIFIED"
    SUPERSEDED = "SUPERSEDED"
    ARCHIVED = "ARCHIVED"


class DatasetInfoResponse(BaseModel):
    """Payload returned by GET /api/v1/system/dataset-info."""
    status: DatasetServingStatus
    dataset_version: str | None = None
    canonical_sha256: str | None = None
    covered_months: list[str] = []
    row_count: int = 0
    unique_projects_count: int | None = None
    source_version_identifier: str | None = None
    ingested_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/system.ts
/**
 * System & Health status types mirroring backend/app/schemas/system.py and health.py
 */

export type DatabaseStatus = "connected" | "disconnected" | "error";

export interface HealthResponse {
  status: "healthy" | "unhealthy" | "degraded";
  database: DatabaseStatus;
  version: string;
  environment: string;
  timestamp: string;
}

export interface DatasetInfoResponse {
  status: "ACTIVE" | "NOT_INGESTED" | "INGESTED_UNVERIFIED" | "SUPERSEDED" | "ARCHIVED";
  dataset_version?: string | null;
  canonical_sha256?: string | null;
  covered_months: string[];
  row_count: number;
  unique_projects_count: number | null;
  source_version_identifier?: string | null;
  ingested_at?: string | null;
}

export interface DatasetServingStatus {
  is_serving: boolean;
  dataset_loaded: boolean;
  message: string;
}
